lib/Sema/TypeCheckDecl.cpp

```diff
@@ -1049,6 +1049,17 @@ static void checkRedeclaration(TypeChecker &tc, ValueDecl *current) {
           continue;
       }
 
+      // If both are VarDecls, and both have exactly the same type, then
+      // matching the Swift 4 behaviour (i.e. just emitting the future-compat
+      // warning) will result in SILGen crashes due to both properties mangling
+      // the same, so it's better to just follow the Swift 5 behaviour and emit
+      // the actual error.
```

Comment:
Can you add a test case for this?

Reply:
The last/unchanged property in the diff for the `overload.swift` is this case.

Reply:
Oops, right, thanks.